clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetFetcherUtils.java

```diff
@@ -240,20 +241,48 @@ Map<TopicPartition, Long> getOffsetResetTimestamp() {
         return offsetResetTimestamps;
     }
 
-    static Map<TopicPartition, OffsetAndTimestamp> buildOffsetsForTimesResult(final Map<TopicPartition, Long> timestampsToSearch,
-                                                                       final Map<TopicPartition, ListOffsetData> fetchedOffsets) {
-        HashMap<TopicPartition, OffsetAndTimestamp> offsetsByTimes = new HashMap<>(timestampsToSearch.size());
+    static <T> Map<TopicPartition, T> buildListOffsetsResult(
```

Comment:
This generic `buildListOffsetsResult` is currently only being used from `buildOffsetsForTimesResult`, was the intention to used it also from `buildOffsetsForTimeInternalResult`?

Reply:
good catch.